include/swift/AST/TypeReprNodes.def

```diff
@@ -46,11 +46,8 @@ TYPEREPR(Error, TypeRepr)
 TYPEREPR(Attributed, TypeRepr)
 
 ABSTRACT_TYPEREPR(DeclRef, TypeRepr)
-  ABSTRACT_TYPEREPR(Ident, DeclRefTypeRepr)
-    TYPEREPR(SimpleIdent, IdentTypeRepr)
-    TYPEREPR(GenericIdent, IdentTypeRepr)
-  TYPEREPR(Member, DeclRefTypeRepr)
-
+  TYPEREPR(UnqualifiedIdent, DeclRefTypeRepr)
+  TYPEREPR(QualifiedIdent, DeclRefTypeRepr)
```

Comment:
I preferred the old names IdentTypeRepr and MemberTypeRepr because {Unq,Q}ualifiedIdentTypeRepr is one more word, but it really doesn't matter.

Reply:
UnqualifiedTypeRepr and QualifiedTypeRepr would work too.

Reply:
This is meant to be a calculated move. In a future where metatypes are modeled as `MemberTypeRepr` or, say, `Any` is modeled as `IdentTypeRepr`, `DeclRefTypeRepr` and `MemberTypeRepr` are misleading names. IMO less specific names, like `IdentTypeRepr` and `QualifiedIdentTypeRepr` respectively, are more coherent. With `IdentTypeRepr` occupied by the current `DeclRefTypeRepr`, the current `IdentTypeRepr` will also need a new name. I thought `UnqualifiedIdentTypeRepr` is OK and would contrast with and complement `QualifiedIdentTypeRepr` nicely.

Reply:
> UnqualifiedTypeRepr and QualifiedTypeRepr would work too.

These are shorter but suggest that arbitrary type representations, not just identifiers, can be qualified.